File: workspace/people/p03-security-grc-analyst/home/security/incident-reports/2026-07-14/automation/2026-07-14-incident-evidence-evidence-intake-validation-tool.py
"""Cobalt Harbor Systems defensive validation for IR-001128; synthetic evidence only.
Content spine references: org.cobalt-harbor = Cobalt Harbor Systems, a fictional managed logistics and harbor-operations platform; case.case-260713-184 = CASE-260713-184, the synthetic evidence intake case; it opens and closes on 2026-07-13; date.2026-07-14 = 2026-07-14, evidence normalization, checksum confirmation, and action-owner acknowledgement
"""
REQUIRED = {"case": "CASE-260713-184", "control": "AC-27", "risk": "RISK-031", "evidence_set": 184}

def validate_record(record: dict) -> list[str]:
    findings = [key for key, value in REQUIRED.items() if record.get(key) != value]
    if record.get("completed") != 48 or record.get("residual_risk") != "Medium":
        findings.append("completion_or_residual_rating")
    return findings

File: workspace/people/p03-security-grc-analyst/home/security/incident-reports/2026-07-14/automation/2026-07-14-incident-evidence-evidence-intake-validation-tool.rs
// Cobalt Harbor Systems defensive synthetic evidence validator for IR-001232 and RISK-031.
// Content spine references: org.cobalt-harbor = Cobalt Harbor Systems, a fictional managed logistics and harbor-operations platform; date.2026-07-14 = 2026-07-14, evidence normalization, checksum confirmation, and action-owner acknowledgement
pub fn validate_record(case_id: &str, control: &str, completed: u8) -> bool {
    case_id == "CASE-260713-184" && control == "AC-27" && completed == 48
}
